order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java

```diff
@@ -70,10 +86,161 @@
 			return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
 		}
 
-//		OrderBean orderBean = orderMapper.toBean(orderWS);
-//		OrderBean order = orderService.createOrder(orderBean);
+		OrderBean orderBean = orderMapper.toBean(orderWS, new CycleAvoidingMappingContext());
+
+		OrderBean order = orderService.createOrder(orderBean);
+
+		return new ResponseEntity<>(orderMapper.toWS(order, new CycleAvoidingMappingContext()), HttpStatus.OK);
+	}
+
+
+	@PutMapping("/order/{uuid}")
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:update")
+	public ResponseEntity<?> updateOrder(@PathVariable("uuid") String orderUuid, @Validated @RequestBody OrderWS orderWS) {
+
+		Optional<OrderBean> tmpOrderBean = orderService.findByUuid(orderUuid);
+
+		if(!tmpOrderBean.isPresent()) {
+			log.error("Error in update order. Fail in finding order uuid={}", orderUuid);
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ORDER_UUID_NOT_FOUND.name(), "Cannot find order by uuid"), HttpStatus.NOT_FOUND);
+		}
+
+		ResponseEntity<?> validateUserAgentUuidResult = validateUserAndAgentUuid(orderWS);
+
+		if(validateUserAgentUuidResult != null) {
+			return validateUserAgentUuidResult;
+		}
+
+		BindException errors = new BindException(orderWS, "OrderWS");
+
+		orderValidator.validate(orderWS, errors);
+
+		if (errors.hasErrors()) {
+			log.error("Error in update order. Fail in update order validation fields. orderWS={}", orderWS.toString());
+			List<ErrorWS> errorWSList = apiErrorsExceptionHandler.generateErrorWSList(errors);
+			return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
+		}
+
+		OrderBean order = tmpOrderBean.get();
+
+		OrderBean orderBean = orderMapper.toBean(orderWS, new CycleAvoidingMappingContext());
+
+		orderBean.setId(order.getId());
```

Comment:
shouldn't need to set the id explicitly. in the order service you can lookup by uuid.

Reply:
I thought search with id is faster than uuid
I can change it now

Reply:
fixed